porch/engine/pkg/engine/options.go

```diff
@@ -46,9 +42,28 @@ func WithCache(cache *cache.Cache) EngineOption {
 	})
 }
 
+func WithBuiltinFunctionRuntime() EngineOption {
+	return EngineOptionFunc(func(engine *cadEngine) error {
+		runtime := newBuiltinRuntime()
+		engine.runtime = runtime
+		return nil
+	})
+}
+
 func WithGRPCFunctionRuntime(address string) EngineOption {
 	return EngineOptionFunc(func(engine *cadEngine) error {
-		runtime, err := createFunctionRuntime(address)
+		runtime, err := newGRPCFunctionRuntime(address)
+		if err != nil {
+			return fmt.Errorf("failed to create function runtime: %w", err)
+		}
+		engine.runtime = runtime
+		return nil
+	})
+}
+
+func WithDelegatingFunctionRuntime(address string) EngineOption {
```

Comment:
we have only few options for function runtimes but consider whether different pattern for constructing them would work better, particularly given the comment above - what if delegating runtime just delegated to another FunctionRuntime...

(I don't have specific desired outcome in mind, just something to consider)

Reply:
Using `MultiRuntime` now